fem/qfunction.hpp

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_QFUNCTION
+#define MFEM_QFUNCTION
+
+#include "../config/config.hpp"
+#include "qspace.hpp"
+#include "gridfunc.hpp"
+
+namespace mfem
+{
+
+/// Represents values or vectors of values at quadrature points on a mesh.
+class QuadratureFunction : public Vector
+{
+protected:
+   QuadratureSpaceBase *qspace; ///< Associated QuadratureSpaceBase object.
+   bool own_qspace; ///< Does this own the associated QuadratureSpaceBase?
+   int vdim; ///< Vector dimension.
+
+public:
+   /// Default constructor, results in an empty vector.
+   QuadratureFunction() : qspace(nullptr), own_qspace(false), vdim(0) { }
```

Comment:
Do we need this? otherwise I would `delete` this default constructor instead.

Reply:
This constructor already exists for `QuadratureFunction` (it's just moved from `gridfunc.hpp`). I agree it would be better deleted, but it's included here for compatibility.